src/customElements/applicationGuidance/Anchor/index.ts

```diff
@@ -0,0 +1,185 @@
+            return;
+        }
+
+        switch (data.type) {
+            case 'application-guidance-anchor-activated':
+                const event = new CustomEvent<QuickFactToggleEventData>('toggle', {
+                    detail: {
+                        anchorId: this.id,
+                        isActive: data.anchorId === this.id,
+                    },
+                });
+                this.dispatchEvent(event);
+                break;
+
+            // case 'application-guidance-active':
+            //     this.isActive = data.isActive;
+            //     this.activeAnchorId = null;
+            //     break;
+        }
+    };
+
+    private applySnugness(rect: ApplicationGuidanceAnchorRect) {
+        if(this.snugFit) {
+            return rect;
+        }
+
+        const snugnessFactor = 16;
+
+        return {
+            top: rect.top - snugnessFactor,
+            left: rect.left - snugnessFactor,
+            width: rect.width + (snugnessFactor * 2),
+            height: rect.height + (snugnessFactor * 2),
+        };
+    }
+
+    private currentRect: ApplicationGuidanceAnchorRect | null = null;
+    private idleCallback: NodeJS.Timeout;
+    private animationFrame: number;
+    private calculateRects = () => {
+        if (this.idleCallback) {
+            window.cancelIdleCallback(this.idleCallback);
+        }
+
+        if (this.animationFrame) {
+            window.cancelAnimationFrame(this.animationFrame);
+        }
+
+        this.idleCallback = window.requestIdleCallback(() => {
+            const rects = [...this.children].map((child) => child.getBoundingClientRect());
+
+            const top = Math.min(...rects.map((r) => r.top));
+            const bottom = Math.max(...rects.map((r) => r.bottom));
+            const left = Math.min(...rects.map((r) => r.left));
+            const right = Math.max(...rects.map((r) => r.right));
+            const rect = this.applySnugness({
+                top,
+                left,
+                width: right - left,
+                height: bottom - top,
+            });
+
+            if (
+                !this.currentRect ||
+                rect.top !== this.currentRect.top ||
+                rect.left !== this.currentRect.left ||
+                rect.width !== this.currentRect.width ||
+                rect.height !== this.currentRect.height
+            ) {
+                this.currentRect = rect;
+                const message: ApplicationGuidanceMessage = {
+                    type: 'application-guide-anchor-rect',
+                    rect,
+                    anchorId: this.id,
+                    scope: this.scope,
+                };
+                window.postMessage(message, window.location.origin);
+            }
+
+            this.animationFrame = window.requestAnimationFrame(this.calculateRects);
+        });
+    };
+
+    render() {
+        return html` <slot></slot> `;
+    }
```

Comment:
this element does not seem to need any shadowroot
`createRenderRoot() {
  return this;
}`